samples/tfm_integration/psa_firmware/README.rst

```diff
@@ -17,52 +17,75 @@ It uses **IPC Mode** for communication, where an IPC mechanism is inserted to
 handle secure TF-M API calls and responses. The OS-specific code to handle
 the IPC calls is in ``tfm_ipc.c``.
 
+TF-M supports three types of firmware upgrade mechanisms:
+``https://tf-m-user-guide.trustedfirmware.org/docs/technical_references/design_docs/tfm_secure_boot.html#firmware-upgrade-operation``
+
+This example uses the overwrite firmware upgrade mechanism, in particular, it showcases
+upgrading the non-secure image which can be built from any other sample in the
+``zephyr/samples`` directory.
+
 The sample prints test info to the console either as a single-thread or
 multi-thread application.
 
+
 Building and Running
 ********************
 
 This project needs another firmware as the update payload. It must use another
 example's hex file, and should be specified on the command line
-as ZEPHYR_FIRMWARE_UPDATE_SAMPLE_BUILD. For example, to use the sample
-`tfm_integration/tfm_ipc` as the payload:
-
-.. code-block:: bash
+as ``CONFIG_APP_FIRMWARE_UPDATE_IMAGE``.
 
-   cd <ZEPHYR_ROOT>
-   west build -p -b lpcxpresso5s69_ns samples/tfm_integration/psa_firmware \
-       -d build/lpcxpresso55s69_ns/tfm_integration/psa_firmware
-       -- -DCONFIG_FIRMWARE_UPDATE_IMAGE=`realpath build/lpcxpresso55s69_ns/tfm_integration/tfm_ipc/zephyr_ns_signed.hex`
-
-
-This project outputs startup status and info to the console. It can be built and
-executed on an ARM Cortex M33 target board or QEMU.
+To use the ``tfm_integration/tfm_ipc`` sample as the NS firmware update
+payload, follow the instructions below:
 
 This sample will only build on a Linux or macOS development system
 (not Windows), and has been tested on the following setups:
 
 - macOS Big Sur using QEMU 6.0.0 with gcc-arm-none-eabi-9-2020-q2-update
-- Linux (NixOS) using QEMU 6.2.50 with gcc from Zephyr SDK 0.13.2
+- Linux (NixOS) using QEMU 6.2.0 with gcc from Zephyr SDK 0.14.1
+- Targets ``MPS3 AN547`` and ``NXP LPCXPRESSO55S69``
 
 On MPS3 AN547:
 ===============
 
-1. Build Zephyr with a non-secure configuration
-   (``-DBOARD=mps3_an547_ns``).
-   Using ``west``
+Build:
+======
+
+1. Build the ``tfm_ipc`` sample with the non-secure board configuration, which will
+generate the firmware image we'll use in ``psa_firmware`` during the update:
 
 .. zephyr-app-commands::
-  :zephyr-app: samples/tfm_integration/psa_firmware
-  :host-os: unix
-  :board: mps3_an547_ns
-  :goals: run
-  :compact:
+   :zephyr-app: samples/tfm_integration/tfm_ipc
+   :host-os: unix
+   :board: mps3_an547_ns
+   :goals: build
+   :build-dir: build/tfm_ipc
+   :compact:
```

Comment:
Is the white space here correct? It looks like there is a difference between the old and new code.